File: exercism/go/archive/clock/clock.go
// Package clock is our add/subtract for a clock
package clock

import "fmt"

// testVersion states which specific suite we work against
const testVersion = 4

// Clock is our basic data structure representing time
type Clock struct {
	hour, min int
}

// New helps instantiate a new Clock struct
func New(hour, minute int) Clock {
	return Clock{hour: 0, min: 0}.Add((hour * 60) + minute)
}

// String is the 'stringer' method for our Clock, returning a string
// representing the time
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.min)
}

// Add is a method for Clock that allows us to add/subtract
// minutes from the current time
func (c Clock) Add(minutes int) Clock {
	oneDay := 60 * 24
	totalMins := ((c.hour * 60) + c.min + minutes) % oneDay

	c.hour = (totalMins / 60) % 24
	c.min = (totalMins % 60)

	if c.hour < 0 {
		c.hour += 24
	}

	if c.min < 0 {
		if c.hour == 0 {
			c.hour = 23
		} else {
			c.hour--
		}
		c.min += 60
	}

	return c
}

File: exercism/src/python/archive/clock/clock.py
""" Module to handle clock input and changes """
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

class Clock(object):
    """ The overarching class that will handle our clock functions """
    def __init__(self, hour, minute):
        """ initialize the clock """
        self._hour = 0
        self._minute = 0

        minutes = (hour * 60) + minute
        self.add(minutes)

    def add(self, minute):
        """ Add the requisite set of minutes to our time """
        one_day = 60 * 24
        total_mins = (self._minute + minute + (self._hour * 60)) % one_day

        self._hour = (total_mins / 60) % 24
        self._minute = total_mins % 60

        if self._hour < 0:
            self._hour += 24

        if self._minute < 0:
            self._minute += 60

            if self._hour == 0:
                self._hour = 23
            else:
                self._hour -= 1

        return "{:02}:{:02}".format(self._hour, self._minute)

    def __str__(self):
        """ How we return data for ourselves """
        return "{:02}:{:02}".format(self._hour, self._minute)

    def __eq__(self, other):
        return self.__dict__ == other.__dict__
